### PR TITLE
Check for nonoverlapping intervals

### PR DESCRIPTION
**Pull request by jdahm**
## Description

Add a check and tests to ensure the AxisIntervals are non-overlapping.

### CODE REVIEW

**Review by egparedes — COMMENTED** [review]
I have a preliminary question before reviewing the PR in detail: hadn't we decided in the GTScript syntax seminar that the intervals should be ordered in the source code according to the forward/backward direction? If that's the case, I think the PR can be redone using a simpler linear, algorithm, since it would be only required to check consecutive intervals.

File: src/gt4py/ir/nodes.py [hunk 1]
```diff
@@ -708,6 +708,25 @@ def full_interval(cls, order=IterationOrder.PARALLEL):
 
         return interval
 
+    def disjoint_from(self, other: "AxisInterval") -> bool:
+        # This made-up constant must be larger than any LevelMarker.offset used
+        DOMAIN_SIZE: int = 1000
+
+        def get_offset(bound: AxisBound) -> int:
+            return (
+                0 + bound.offset if bound.level == LevelMarker.START else DOMAIN_SIZE + bound.offset
```
Comment (egparedes) [inline src/gt4py/ir/nodes.py:717]: Kind of nit-picking,  but I would use a more meaningful constant value for comparing values starting from the end:

```suggestion
                0 + bound.offset if bound.level == LevelMarker.START else sys.maxsize + bound.offset
```

**Review by egparedes — COMMENTED** [review]
It looks good, just have one comment.

File: src/gt4py/ir/nodes.py (hunk 1 above)
Comment (jdahm) [reply src/gt4py/ir/nodes.py:717]: Can you add to `sys.maxsize` consistently without overflow?

**Review by egparedes — APPROVED** [review]
Good to go now! Thanks for your contribution.

**Pull request merged**